### PR TITLE
[FIX] dev:di:info duplicates plugin info

### PR DESCRIPTION
**Pull request by coderimus**
### Description
Hello,

Looks like the execution of `dev:di:info` duplicates info about plugins by for the or the Preference if the same plugin class and its type were introduced in plugins by class name section.

For example, execute `php bin/magento dev:di:info "Magento\Sales\Model\Order"
` and the output result will be
image

with provided fix it will look like this

image

The problem is seems to be in this method: 
```php
    private function addPluginToList($pluginInstance, $pluginMethod, $methodTypes, $typeCode)
    {
        if ($methodTypes & $typeCode) {
            if (!array_key_exists($pluginInstance, $this->pluginList[$this->pluginTypeMapping[$typeCode]])) {
                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance] = [];
            }
            $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod ;
        }
    }
```
It checks whether value exists in array or not. And if it exists it will be added +1 time. But it may logically to add an item once if it not exists.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios
execute `dev:di:info` command for class and check "Plugins for the Preference:" section. Seems to have duplicated info.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=16474) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @Coderimus. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Developer/Model/Di/PluginList.php [hunk 1]
```diff
@@ -143,9 +143,8 @@ private function addPluginToList($pluginInstance, $pluginMethod, $methodTypes, $
     {
         if ($methodTypes & $typeCode) {
             if (!array_key_exists($pluginInstance, $this->pluginList[$this->pluginTypeMapping[$typeCode]])) {
-                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance] = [];
+                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod;
             }
-            $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod ;
```
Comment (orlangur) [inline app/code/Magento/Developer/Model/Di/PluginList.php]:
```
$this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod ; ->

$this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][$pluginMethod] = 1
```

Please avoid duplication by reversing array. Please check if it is possible to remove `if` while avoiding PHP notice.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Developer/Model/Di/PluginList.php [hunk 2]
```diff
@@ -143,6 +143,10 @@ private function addPluginToList($pluginInstance, $pluginMethod, $methodTypes, $
     {
         if ($methodTypes & $typeCode) {
             if (!array_key_exists($pluginInstance, $this->pluginList[$this->pluginTypeMapping[$typeCode]])) {
+                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance] = [];
+            }
+
+            if (!in_array($pluginMethod, $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance])) {
```
Comment (coderimus) [inline app/code/Magento/Developer/Model/Di/PluginList.php]: Hi @orlangur ,
What do you think about this modification? I implemented this second condition just because it looks like a cheapest and still technologically possible solution. Also, this method is totally isolated and used only for this command.

Comment (orlangur) [reply app/code/Magento/Developer/Model/Di/PluginList.php]: Too wordy and not elegant as to me but still good enough :) Asymptotic  complexity became poor but should not be noticeable.

Good job with tiniest possible diff 👍

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-2177](https://jira.corp.magento.com/browse/ENGCOM-2177) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @Coderimus. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.6. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.
